torchvision/utils.py

```diff
@@ -121,10 +126,74 @@ def save_image(
             If a file object was used instead of a filename, this parameter should always be used.
         **kwargs: Other arguments are documented in ``make_grid``.
     """
-    from PIL import Image
     grid = make_grid(tensor, nrow=nrow, padding=padding, pad_value=pad_value,
                      normalize=normalize, range=range, scale_each=scale_each)
     # Add 0.5 after unnormalizing to [0, 255] to round to nearest integer
     ndarr = grid.mul(255).add_(0.5).clamp_(0, 255).permute(1, 2, 0).to('cpu', torch.uint8).numpy()
     im = Image.fromarray(ndarr)
     im.save(fp, format=format)
+
+
+def draw_bounding_boxes(
+    image: torch.Tensor,
+    boxes: torch.Tensor,
+    labels: torch.Tensor,
+    label_names: List[str] = None,
+    colors: Dict[int, str] = None,
+    draw_labels: bool = True,
+    width: int = 1
+) -> torch.Tensor:
+
+    """
+    Draws bounding boxes on given image.
+
+    Args:
+        image (Tensor): Tensor of shape (C x H x W) or (1 x C x H x W)
+        bboxes (Tensor): Tensor of size (N, 4) containing bounding boxes in (xmin, ymin, xmax, ymax) format.
+        labels (Tensor): Tensor of size (N) Labels for each bounding boxes.
+        label_names (List): List containing labels excluding background.
+        colors (dict): Dict with key as label id and value as color name.
+        draw_labels (bool): If True (default) draws label names on bounding boxes.
+        width (int): Width of bounding box.
+    """
+
+    # Code co-contributed by sumanthratna
+
+    if not (torch.is_tensor(image)):
+        raise TypeError(f'tensor expected, got {type(image)}')
+
+    if(image.dim() == 4):
+        if(image.shape[0] == 1):
+            image = image.squeeze(0)
+        else:
+            raise ValueError("Batch size > 1 is not supported. Pass images with batch size 1 only")
+
+    if label_names is not None:
+        # Since for our detection models class 0 is background
+        label_names.insert(0, "__background__")
```

Comment:
I'm not really super comfortable adding this here. I think that this is something that the users should specify themselves.

Plus, this modifies the passed list inplace, and the second invocation will already have `__background__` added.

Reply:
Let's leave this to user. I guess user should probably know that `background` is 0.

The other way I was thinking was to pass `dict()` instead of list which specifies class name and class number. (I feel it can be naturally ordered by indexes in list though)


I'm removing the background for now.